### ISSUE TITLE
modal_location_from_dates hangs when no data for end date

### ISSUE THREAD

**Issue by josh-gree** [Flowminder/FlowKit#579]
**Product**
FlowClient

**Version**
Flowclient '0.5.1'

**To Reproduce**
```
ml = modal_location_from_dates('2016-01-02','2016-01-03',method="last",aggregation_unit="admin2")
res_id = run_query(conn, ml)
```

When no data for `2016-01-03` hangs indefinitely.

**Comment by greenape** [comment]
This should be fixed as of #574, but reopen if you still get it with latest master.

**Comment by greenape** [comment]
#514 is this also
